### DISCUSSION TITLE
e_MOD_REWRITE question

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5360] [Q&A Core]
Hi,
I have a site, no important changes from 2.3.3 maybe, nothing I am aware of.  Some fixes from 2.4 ? The site was working normally long time. PHP 8.1. 

For some reason, I got a message about rewrite mode and I had to use 
`define('e_MOD_REWRITE', true);`
in e107_config.php
It wasn't this way before. 

I am running more e107 installation with the same provider, I had no similar issue - never, so something was changed just for this domain.

When I check php.ini, the only related values to rewrite is 
![image](https://github.com/user-attachments/assets/85f37672-2373-47d1-9208-10c2febfd9d2)

There is no warning in e107 PHP info part about PHP configuration.
The only notice was in the Preferences/URL configuration. 

What could be wrong?  Where to look. 

Thanks

**Comment by Jimmi08** [comment]
I just found that my other site (e107sk.com) has the same issue.  Cache clearing was needed after changing e107_config.

**Comment by Vodhin** [comment]
rewrite rules are defined in .htaccess and have to do with how the server handles url requests that match those rules.

**Reply by Jimmi08** [reply]
.htaccess wasn't changed, it is the default one generated by e107, used for years, I have never issue with this before. 

What I should check on the server side?  I mean - something what I can check myself at my hosting administration or what I should ask my hosting provider. 

I wouldn't find this but users reported not working downloads so it can't be long time and I am sure I haven't changed / updated files.

**Reply by Vodhin** [reply]
Honestly, I am the last person to ask about rewrite rules: I barely understand the syntax myself. I would think that the culprit is something in the v2.4 version that is encoding a server request in a way that fails when passes through the old .htaccess file. if `define('e_MOD_REWRITE', true);` solves the issue, then you should be fine.

**Comment by Moc** [comment]
In `PHP Info` / php.ini, `mod_rewrite` should be listed under `Loaded Modules`. If that's not the case, then mod_rewrite is not configured properly on your server. 

The  "url_rewriter" items on your screenshot are not related to mod_rewrite.
